packages/devtools_app/lib/src/shared/config_specific/launch_url/launch_url.dart

```diff
@@ -11,14 +11,20 @@ import '_launch_url_desktop.dart'
 Future<void> launchUrl(String url) async {
   final parsedUrl = Uri.tryParse(url);
 
-  if (parsedUrl != null && await url_launcher.canLaunchUrl(parsedUrl)) {
-    await url_launcher.launchUrl(parsedUrl);
-  } else {
-    notificationService.push('Unable to open $url.');
+  try {
+    if (parsedUrl != null && await url_launcher.canLaunchUrl(parsedUrl)) {
+      await url_launcher.launchUrl(parsedUrl);
+    } else {
+      notificationService.push('Unable to open $url.');
+    }
+  } finally {
+    // Always pass the request up to VS Code because we could fail both silently
+    // (the usual behaviour) or with another error like
+    // "Attempted to call Window.open with a null window"
+    // https://github.com/flutter/devtools/issues/6105.
+    //
+    // In the case where we are not in VS Code, there will be nobody listening
+    // to the postMessage this sends.
+    launchUrlVSCode(url);
   }
```

Comment:
for example, what if at some point, `url_launcher.launchUrl` starts working for the embedded environment? then we'd be launching the url twice.

Reply:
We don't expect `launchUrlVSCode` to fail, it will just silently do nothing (it'll `postMessage` but nobody is listening).

We can check that though - is there any possibility of ending up here without it being on `globals`? (I don't think so, but it feels slightly riskier than the change to just call this when an exception occurs for a cherry-pick)

Reply:
Just to be clear - the `launchUrlVSCode` method does not throw, the issue was that `url_launcher.launchUrl` would sometimes throw (in the previous version, not the current stable).

This code was originally written to always call `launchUrlVSCode`, but in the case where `url_launcher.launchUrl` throws (something we never expected but was happening in the previous stable), it would be skipped.

My change here just moves `launchUrlVSCode` into a `finally` block so that it will be called _also_ if `url_launcher.launchUrl` throws, and not only if it completed without exception.

I didn't use a `catch` because I didn't want to mask the original (unexpected) exception if it comes back, but this way we'll at least still open the URL from VS Code instead of silently doing nothing.